db/records.py

```diff
@@ -47,8 +53,12 @@ def _execute_query(query, engine):
 
 
 def get_record(table, engine, id_value):
-    primary_key_column = _get_primary_key_column(table)
-    query = select(table).where(primary_key_column == id_value)
+    primary_key_columns = _get_primary_key_columns(table)
+
+    if len(primary_key_columns) != 1:
+        raise NotUniquePrimaryKey("This table does not have a unique primary key.")
+
+    query = select(table).where(primary_key_columns[0] == id_value)
```

Comment:
Please use `_get_primary_key_column` here, that should be the only way we access a singular primary key.